src/main/java/edu/umich/eecs/april/apriltag/ApriltagDetectorActivity.java

```diff
@@ -69,63 +81,21 @@ protected void onCreate(Bundle savedInstanceState) {
                     new String[]{Manifest.permission.CAMERA},
                     MY_PERMISSIONS_REQUEST_CAMERA);
         } else {
-            this.has_camera_permissions = 1;
+            startCamera();
         }
     }
 
-    /** Release the camera when the application is exited */
     @Override
     protected void onDestroy() {
         super.onDestroy();
-        stopThreads();
-        Log.i(TAG, "Finished destroying.");
-    }
-
-    /** Release the camera when application focus is lost */
-    protected void onPause() {
-        super.onPause();
-        stopThreads();
-        Log.i(TAG, "Finished pause.");
+        detectorExecutor.shutdown();
     }
 
-    private void stopThreads() {
-        if (mCameraPreviewThread != null) {
-            mCameraPreviewThread.interrupt();
-            mCameraPreviewThread.destroy();
-            try {
-                mCameraPreviewThread.join();
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-            mCameraPreviewThread = null;
-        }
-        if (mDetectionThread != null) {
-            mDetectionThread.interrupt();
-            mDetectionThread.destroy();
-            try {
-                mDetectionThread.join();
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-            mDetectionThread = null;
-        }
-    }
-
-    /** (Re-)initialize the camera */
-    protected void onResume() {
-        super.onResume();
-
-        // Check permissions
-        if (this.has_camera_permissions == 0) {
-            Log.w(TAG, "Missing camera permissions.");
-            return;
-        }
-
-        // DETECTION INIT
+    private void applyPreferences() {
         // Re-initialize the Apriltag detector as settings may have changed
         verifyPreferences();
         SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
-        double decimation = Double.parseDouble(sharedPreferences.getString("decimation_list", "8"));
+        double decimation = Double.parseDouble(sharedPreferences.getString("decimation_list", "2"));
```

Comment:
The ImageAnalyzer API returns a 640x480 image, and we already discussed offline to add a resolution selector. It's more responsive to have the hardware set resolution instead of using software decimation.

One way to think about this setting is "how many times more pixels do we need to quad detect vs decode." A setting between 2-4 seems right for this, while letting the user adjust resolution for the detection range / computation time tradeoff.

Reply:
👍 can you make sure there's a ticket for adding resolution selector?